src/FrameSemantics.cc

```diff
@@ -282,289 +282,511 @@ static Errors resolveModelPoseWithPlacementFrame(
   return errors;
 }
 
-/////////////////////////////////////////////////
-Errors buildFrameAttachedToGraph(
-    ScopedGraph<FrameAttachedToGraph> &_out, const Model *_model, bool _isRoot)
+/// \brief Base struct the contains a few common members. These structs provide
+/// a common API used by the build*Graph functions to access the various
+/// attributes and retrieve children of regular DOM objects and Interface
+/// Elements.
+struct WrapperBase
 {
-  Errors errors;
-
-  if (!_model)
-  {
-    errors.push_back({ErrorCode::ELEMENT_INVALID,
-        "Invalid sdf::Model pointer."});
-    return errors;
-  }
-  else if (!_model->Element())
+  /// \brief Name of the entity
+  const std::string name;
+  /// \brief Element type, such as Model, Link, or Interface Frame.
+  const std::string elementType;
+  /// \brief Frame type used int the FrameAttachedTo or PoseRelativeTo graphs.
+  const FrameType frameType;
+};
+
+/// \brief Wrapper for sdf::Link and sdf::InterfaceLink
+struct LinkWrapper: public WrapperBase
+{
+  /// \brief Constructor that takes an sdf::Link
+  explicit LinkWrapper(const sdf::Link &_link)
+      : WrapperBase{_link.Name(), "Link", FrameType::LINK},
+        rawPose(_link.RawPose()),
+        relativeTo(_link.PoseRelativeTo()),
+        finalRelativeTo(relativeTo)
   {
-    errors.push_back({ErrorCode::ELEMENT_INVALID,
-        "Invalid model element in sdf::Model."});
-    return errors;
   }
-  else if (_model->LinkCount() == 0 && _model->ModelCount() == 0 &&
-      _model->InterfaceModelCount() == 0 && !_model->Static())
+
+  /// \brief Constructor that takes an sdf::InterfaceLink
+  explicit LinkWrapper(const sdf::InterfaceLink &_ifaceLink)
+      : WrapperBase{_ifaceLink.Name(), "Interface Link", FrameType::LINK},
+        rawPose(_ifaceLink.PoseInModelFrame()),
+        relativeTo("__model__"),
+        finalRelativeTo(relativeTo)
   {
-    errors.push_back({ErrorCode::MODEL_WITHOUT_LINK,
-                     "A model must have at least one link."});
-    return errors;
   }
 
-  auto frameType =
-      _model->Static() ? sdf::FrameType::STATIC_MODEL : sdf::FrameType::MODEL;
-
-  const std::string scopeContextName = "__model__";
+  /// \brief Raw pose of the entity.
+  const ignition::math::Pose3d rawPose;
+  /// \brief The //pose/@relative_to attribute.
+  const std::string relativeTo;
```

Comment:
if `relativeTo` is always a copy of the attribute, how about calling it `rawRelativeTo` to mirror `rawPose`. then we can change `finalRelativeTo` to `relativeTo`

Reply:
Thanks for the suggestion. Done in https://github.com/ignitionrobotics/sdformat/pull/764/commits/59050f96134d6bf0cea07e35ab7916b1da7b70d9